definitions/3.0.0/anySchema.json

```diff
@@ -1,16 +1,21 @@
 {
-  "if": {
-    "required": [
-      "schema"
-    ]
-  },
-  "then": {
-    "$ref": "http://asyncapi.com/definitions/3.0.0/multiFormatSchema.json"
-  },
-  "else": {
-    "$ref": "http://asyncapi.com/definitions/3.0.0/schema.json"
-  },
-  "description": "An object representing either a schema or a multiFormatSchema based on the existence of the 'schema' property. If the property 'schema' is present, use the multi-format schema. Use the default AsyncAPI Schema otherwise.",
+  "$id": "http://asyncapi.com/definitions/3.0.0/anySchema.json",
   "$schema": "http://json-schema.org/draft-07/schema#",
-  "$id": "http://asyncapi.com/definitions/3.0.0/anySchema.json"
+  "type": "object",
+  "description": "An object representing either a Reference, a Schema or a Multi Format Schema",
+  "oneOf": [
+    {
+      "description": "Because of $ref collision in Reference and AsyncAPI Schema(includes $ref from Json Schema)",
+      "not": {"required": ["schemaFormat", "schema"]},
+      "$ref": "http://asyncapi.com/definitions/3.0.0/schema.json"
+    },
+    {
+      "type": "object",
+      "required": ["schemaFormat", "schema"],
+      "not": {"required": ["$ref"]},
+      "minProperties": 2,
+      "maxProperties": 2,
+      "$ref": "http://asyncapi.com/definitions/3.0.0/multiFormatSchema.json"
+    }
+  ]
```

Comment:
Meant the entire thing @Pakisan, `anySchema` does not need any changes to work 🙂